src/main/scala/djinni/Main.scala

```diff
@@ -460,8 +451,16 @@ object Main {
       cWrapperBaseLibIncludePrefix,
       pyImportPrefix)
 
-
+    // Resolve names in IDL file, check types.
     try {
+      resolver.resolve(meta.defaults, idl, outSpec) match {
+        case Some(err) =>
+          System.err.println(err)
+          System.exit(1); return
+        case _ =>
+      }
```

Comment:
This block of code was moved down so we could use the information from `outSpec`. (I'm not 100% sure about this one.)

Reply:
Only solution I see for this is to pass all `<lang>outFolder` parameters to `resolve` one by one?

```
resolver.resolve(meta.defaults, idl, javaOutRequired: javaOutFolder.isDefined, objcOutRequired: objcOutFolder.isDefined, objcppOutRequired: objcppOutFolder.isDefined, cppcliOutRequired: cppcliOutFolder.isDefined, cppOutRequired: cppOutFolder.isDefined)
```

One advantage I could see in this approach is that the `metaFromYaml` internals wouldn't be as verbose, because the `.isDefined` check only happens once in the function call:

```
MExtern.Cpp(
      nested(td, "cpp")("typename").toString,
      nested(td, "cpp")("header").toString,
      nested(td, "cpp")("byValue").asInstanceOf[Boolean]),
      nested(td, cppOut, "cpp", "typename", _.toString),
      nested(td, isRequired: cppOutRequired, "cpp", "header", _.toString),
      nested(td, isRequired: cppOutRequired, "cpp", "byValue", _.asInstanceOf[Boolean])),
```

Plus the code could be a bit easier to understand because the input for `resolve()` is defined more precise.

I'm sure you have thought about that for a longer time than me though. Did you consider this? The annoying part about it is that all parameters have to be passed to resolved & then passed to `metaFromYaml`.

Reply:
I had thought about passing only the parts of `outSpec` that were necessary (e.g. `cppOutDir`) and call `isDefined` on them "in the last moment", but it didn't feel right. Defining the parameters to `resolve()` that way -- `<lang>OutRequired` -- definitely makes much more sense. I'd probably even leave the "out" part out: `langRequired`. Let me give it a shot and we see. Thanks for the suggestion!